source/adaptive_load/plugin_util.cc

```diff
@@ -0,0 +1,75 @@
+#include "envoy/config/core/v3/base.pb.h"
+
+#include "nighthawk/adaptive_load/input_variable_setter.h"
+#include "nighthawk/adaptive_load/metrics_plugin.h"
+#include "nighthawk/adaptive_load/scoring_function.h"
+#include "nighthawk/adaptive_load/step_controller.h"
+
+#include "external/envoy/source/common/common/statusor.h"
+#include "external/envoy/source/common/config/utility.h"
+
+namespace Nighthawk {
+
+absl::StatusOr<InputVariableSetterPtr>
+LoadInputVariableSetterPlugin(const envoy::config::core::v3::TypedExtensionConfig& config) {
+  try {
+    auto& config_factory = Envoy::Config::Utility::getAndCheckFactoryByName<InputVariableSetterConfigFactory>(config.name());
+    absl::Status validation_status = config_factory.ValidateConfig(config.typed_config());
+    if (!validation_status.ok()) {
+      return validation_status;
+    }
+    return config_factory.createInputVariableSetter(config.typed_config());
+  } catch (const Envoy::EnvoyException& e) {
+    return absl::InvalidArgumentError(
+        absl::StrCat("Could not load plugin: ", config.DebugString(), ": ", e.what()));
```

Comment:
could this log line accidentally bleed sensitive information into the logs? (may also apply below)

Reply:
Good point, this will end up in the log later -- changed to name().